clang/lib/Sema/SemaChecking.cpp

```diff
@@ -713,6 +713,11 @@ static ExprResult SemaBuiltinDumpStruct(Sema &S, CallExpr *TheCall) {
     return ExprError();
   }
   const RecordDecl *RD = PtrArgType->getPointeeType()->getAsRecordDecl();
+  if (!RD->isCompleteDefinition()) {
```

Comment:
You need to use `RequireCompleteType` here to properly trigger instantiation of templated classes and handle definition visibility checking for modules.

Reply:
Thank you! Added the template case to the test.